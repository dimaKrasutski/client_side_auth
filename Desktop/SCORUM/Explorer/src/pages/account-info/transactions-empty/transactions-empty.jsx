import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';
import { IconTransactionsEmpty } from 'theme/components/icons/icon-transactions-empty/icon-transactions-empty';

export const TransactionsEmpty = ({ titleId, subtitleId, values }) => (
  <div className="text-center pt-3_5 pt-sm-8_5 text-gray-dark-secondary">
    <IconTransactionsEmpty />
    <h4 className="mt-0_5 mb-0">
      <FormattedMessage id={titleId} />
    </h4>
    {
      subtitleId &&
      <FormattedHTMLMessage id={subtitleId} values={values} />
    }
  </div>
);

TransactionsEmpty.propTypes = {
  titleId: PropTypes.string,
  subtitleId: PropTypes.string,
  values: PropTypes.shape().isRequired,
};

TransactionsEmpty.defaultProps = {
  titleId: 'explorer.account-info.transactions.empty.title',
  subtitleId: 'explorer.account-info.transactions.empty.subtitle',
};
