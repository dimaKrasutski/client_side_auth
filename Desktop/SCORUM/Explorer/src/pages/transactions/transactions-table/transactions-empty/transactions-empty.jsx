import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedHTMLMessage } from 'react-intl';
import { IconTransactionsEmpty } from 'theme/components/icons/icon-transactions-empty/icon-transactions-empty';

export const TransactionsEmpty = ({ titleId, subtitleId }) => (
  <div className="text-center pt-3_5 pt-sm-8_5 text-gray-dark-secondary">
    <IconTransactionsEmpty />
    <h4 className="mt-0_5 mb-0">
      <FormattedMessage id={titleId} />
    </h4>
    <FormattedHTMLMessage id={subtitleId} />
  </div>
);

TransactionsEmpty.propTypes = {
  titleId: PropTypes.string,
  subtitleId: PropTypes.string,
};

TransactionsEmpty.defaultProps = {
  titleId: 'explorer.transactions.empty.title',
  subtitleId: 'explorer.transactions.empty.subtitle',
};
