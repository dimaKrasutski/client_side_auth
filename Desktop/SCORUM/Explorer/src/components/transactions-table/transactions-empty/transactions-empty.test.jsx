import React from 'react';
import { TransactionsEmpty } from './transactions-empty';

describe('<TransactionsEmpty />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<TransactionsEmpty titleId="test-title" subtitleId="test-subtitle" />);
    expect(wrapper).toMatchSnapshot();
  });
});
