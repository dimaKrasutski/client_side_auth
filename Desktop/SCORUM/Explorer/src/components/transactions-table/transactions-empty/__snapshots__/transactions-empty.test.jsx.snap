// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionsEmpty /> should match snapshot 1`] = `
<div
  className="text-center pt-3_5 pt-sm-8_5 text-gray-dark-secondary"
>
  <IconTransactionsEmpty />
  <h4
    className="mt-0_5 mb-0"
  >
    <FormattedMessage
      id="test-title"
      values={Object {}}
    />
  </h4>
  <FormattedHTMLMessage
    id="test-subtitle"
    values={Object {}}
  />
</div>
`;
